nilearn/input_data/__init__.py

```diff
@@ -1,11 +1,31 @@
 """
-The :mod:`nilearn.input_data` module includes scikit-learn tranformers.
+The :mod:`nilearn.input_data` module used to include masker objects.
+It is deprecated since release 0.8.2 in favor of the
+:mod:`~nilearn.maskers` module.
+
+Please consider updating your code:
+
+.. code-blocks::python
+    from nilearn.input_data import NiftiMasker
+
+becomes:
+
+.. code-blocks::python
+    from nilearn.maskers import NiftiMasker
+
+Note that all imports that used to work will continue to do so with
+a simple warning.
```

Comment:
```suggestion
a simple warning until 0.11.0.
```

Maybe specify the deprecation cycle length here? I think it was two "major" releases, but I could be wrong.

Reply:
Good question. I think the plan was to keep these imports working (ping @jeromedockes). 
If we go for a deprecation cycle, I'd vote for a longer one.

Reply:
yes I think the old imports should keep working forever or at least for a very long time. the size of this PR (87 files) gives a hint as to how annoying it will be for users to change all their imports of the `input_data` module (and the situation is even worse for client code that will have to support both new and old nilearn versions)

Reply:
But I agree with @tsalo  that we should make a decision about that so we can be more explicit and tell users until when existing code will keep working

Reply:
Fair enough. What about doubling the usual deprecation cycle length? This would lead us to release 0.13.0.

Reply:
how about "at least until release 0.13.0" ?

Reply:
Sounds good to me! Added in dacf94c